### PR TITLE
Add parameter `preserve_index` to `from_pandas`

### PR DESCRIPTION
**Pull request by Sorrow321**
Added optional parameter, so that user can get rid of useless index preserving. [Issue](https://github.com/huggingface/datasets/issues/3563)

### CODE REVIEW

**Comment by Sorrow321** [comment]
> 

I did `make style` and it affected over 500 files
image

**Comment by Sorrow321** [comment]
Nvm I was using wrong black version

File: src/datasets/arrow_dataset.py [hunk 1]
```diff
@@ -760,6 +761,7 @@ def from_pandas(
             features (:class:`Features`, optional): Dataset features.
             info (:class:`DatasetInfo`, optional): Dataset information, like description, citation, etc.
             split (:class:`NamedSplit`, optional): Name of the dataset split.
+            preserve_index (:obj:`bool`, optional): Whether to preserve pd.DataFrame index while constructing pyarrow.Table. By default, it only preserves custom index.
```
Comment (lhoestq) [inline src/datasets/arrow_dataset.py:764]: (nit) maybe we can also add more details from pyarrow.Table.from_pandas:
```suggestion
            preserve_index (:obj:`bool`, optional): Whether to store the index as an additional column in the resulting Dataset.
                The default of None will store the index as a column, except for RangeIndex which is stored as metadata only.
                Use preserve_index=True to force it to be stored as a column.
```

**Review by lhoestq — APPROVED** [review]
Perfect thanks :)

**Pull request merged**